### PR TITLE
Fixed tests on newest node (tested on v10.15 and some v11's)

### LINKED ISSUE #19: Fix tests not working on Node >9
Mock-FS is not supported on Node >9: https://github.com/tschaub/mock-fs/issues/254#issuecomment-425573723

The solutions would be to remove the tests using filesystem or encapsulate the filesystem interface somehow / use alternative solution to mock-fs.

### PR DESCRIPTION
**Pull request by cymerrad**
Fixes #19 

## Changelog
- Testing framework changed to Jest
- Using memfs for mocks
- Reduced dependencies count
  
However, I think that that efficacy of tests may have been lost in the process...  
I haven't given much thought to how `defaultContext()` should work when combined with forced `process.chdir("...")`. I.e. I'd like you to check, if tests still make sense and do actually test the resolving business.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -8,35 +8,32 @@
   "license": "LGPL-3.0-or-later",
   "scripts": {
     "build": "tsc",
-    "test": "mocha build/test/**/*.test.js",
+    "test": "yarn jest",
```
Comment (ritave) [inline package.json]: Just `jest` is enough. The scripts could be run from eg. `npm` without yarn installed

File: package.json [hunk 2]
```diff
@@ -8,35 +8,32 @@
...
+    "jest": "^23.6.0",
     "nock": "^9.3.2",
     "rimraf": "^2.6.2",
+    "ts-jest": "^23.10.5",
     "typescript": "^2.8.1"
   },
   "dependencies": {
     "debug": "^3.1.0",
     "hosted-git-info": "^2.6.0",
+    "memfs": "^2.14.2",
```
Comment (ritave) [inline package.json]: Move to devDependencies (tests only). `yarn remove memfs; yarn add -d memfs`

File: package.json [hunk 3]
```diff
@@ -8,35 +8,32 @@
...
     "request": "^2.85.0",
     "tmp": "^0.0.33"
   },
   "engines": {
-    "node": "<10.5"
+    "node": ">=10.0.0"
```
Comment (ritave) [inline package.json]: The tests probably work on Node 9. Remove the `engines` object altogether

File: test/__tests__/fsresolver.spec.ts [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+      instance = FsResolver();
+    });
+
+    it("finds absolute paths", async function () {
+      vol.fromJSON({
+        "/file.test": "correct",
+        "./file.test": "wrong",
+      });
+
+      const paf = await instance("/file.test", defaultContext());
```
Comment (ritave) [inline test/__tests__/fsresolver.spec.ts]: use a more descriptive name that works in english

File: test/__tests__/fsresolver.spec.ts [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+    contentsParser = FsParser();
+    process.chdir(__dirname);
+  });
+
+  afterEach(function () {
+    vol.reset();
+  });
+
+  describe("without root prefix", function () {
+    beforeEach(function () {
```
Comment (ritave) [inline test/__tests__/fsresolver.spec.ts]: The `beforeEach` is the same for both root prefix and without root prefix. Move both upwards to the main `describe`

Comment (ritave) [reply test/__tests__/fsresolver.spec.ts]: Also might be worth moving into `beforeAll`

File: package.json [hunk 6]
```diff
@@ -8,35 +8,32 @@
...
   }
-}
+}
```
Comment (squadack) [inline package.json]: no newline at the end of file

File: test/__mocks__/fs.js [hunk 7]
```diff
@@ -0,0 +1,2 @@
+process.chdir('/');
+module.exports = require('memfs');
```
Comment (squadack) [inline test/__mocks__/fs.js]: newline

File: tsconfig.json [hunk 8]
```diff
@@ -17,5 +17,8 @@
     "noImplicitReturns": true,
     "esModuleInterop": true
   },
-  "include": ["src/**/*.ts", "examples/**/*.ts", "test/**/*.ts"]
-}
+  "include": [
+    "src/**/*.ts",
+    "examples/**/*.ts"
+  ]
+}
```
Comment (squadack) [inline tsconfig.json]: newline missing

File: package.json (hunk 3 above)
Comment (cymerrad) [reply package.json]: Yeah, but I'd really like to dissuade anyone from using those awkward, in-between-LTS versions of node.

File: test/__tests__/fsresolver.spec.ts (hunk 4 above)
Comment (cymerrad) [reply test/__tests__/fsresolver.spec.ts]: Actually SOMEONE introduced a collision of names in this very spec.
`path` was a variable and a module name. I just thought it was funny and who-cares-eh

File: package.json (hunk 3 above)
Comment (ritave) [reply package.json]: @cymerrad That's not up to us to decide, engine definitions might disable using our package altogether even if it works fine - http://www.marcusoft.net/2015/03/packagejson-and-engines-and-enginestrict.html

File: package.json (hunk 2 above)
Comment (cymerrad) [reply package.json]: Shouldn't also the `tmp` package be in devDeps?

Comment (cymerrad) [reply package.json]: And `request`? They both are used for resolving.

Comment (ritave) [reply package.json]: Memfs is used only for tests, while tmp and request are both used by the end-user when they resolve something.

**Pull request merged**